src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_upscale.py

```diff
@@ -90,6 +92,25 @@ def __init__(
     ):
         super().__init__()
 
+        # check if vae has a config attribute `scaling_factor` and if it is set to 0.08333, else set it to 0.08333 and deprecate
+        is_vae_scaling_factor_set_to_0_08333 = (
+            hasattr(vae.config, "scaling_factor") and vae.config.scaling_factor == 0.08333
+        )
+        if not is_vae_scaling_factor_set_to_0_08333:
+            deprecation_message = (
+                "The configuration file of the vae has  set the configuration `scaling_factor` to"
+                f" {vae.config.scaling_factor}. which seems highly unlikely. If your checkpoint is a fine-tuned"
+                " version of any of the following: \n- stabilityai/stable-diffusion-x4-upscaler \n you should change"
+                " 'scaling_factor' to 0.08333 in the configuration file. Please make sure to update the config"
+                " accordingly as not setting `scaling_factor` in the config might lead to incorrect results in future"
+                " versions. If you have downloaded this checkpoint from the Hugging Face Hub, it would be very nice"
+                " if you could open a Pull request for the `vae/config.json` file"
+            )
+            deprecate("wrong scaling_factor", "1.0.0", deprecation_message, standard_warn=False)
+            new_config = dict(vae.config)
+            new_config["scaling_factor"] = 0.08333
+            vae._internal_dict = FrozenDict(new_config)
```

Comment:
special case.

But actually thinking more about it, this way of adding a new argument, and deprecating in the same PR seems wrong.

Reply:
I think it should be ok as long as we add the parameter to `stabilityai/stable-diffusion-x4-upscaler` in the repo. Maybe we can open PRs for forked models too? Perhaps we can change the warning a little bit to clarify that the parameter might be absent from the json file and not set to a wrong value.